### PR TITLE
Add basic ESLint config

### PR DESCRIPTION
**Pull request by pdehaan**

### CODE REVIEW

File: electron.js [hunk 1]
```diff
@@ -1,39 +1,39 @@
-'use strict'
-const electron = require('electron')
+'use strict';
```
Comment (pdehaan) [inline electron.js]: Not 100% sure what this file is, and if we should be modifying it or if it should be part of the .eslintignore file so it doesn't get linted/formatted.

File: encryption.js [hunk 2]
```diff
@@ -60,34 +60,37 @@ function joinIvAndData(iv, data) {
...
 /* Encryption using email, passphrase and content */
 function encrypt(passphrase, content) {
-  let initVector = new Uint8Array(ivLen);
+  const initVector = new Uint8Array(ivLen);
   crypto.getRandomValues(initVector);
 
   const passphraseKey = encoder.encode(passphrase);
   const data = encoder.encode(content);
 
+  // eslint-disable-next-line no-console
```
Comment (pdehaan) [inline encryption.js]: I ended up adding this in a few places. Not thrilled about it, but I didn't want to be so bold that I deleted all the debugging `console.log()` statements.

File: www/js/index.js [hunk 3]
```diff
@@ -16,31 +16,32 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-var app = {
+const app = {
```
Comment (pdehaan) [inline www/js/index.js]: Same with this file. Not sure if I should be editing "www/js/index.js" or if we add it to .eslintignore so ESLint ignores it.

**Comment by pdehaan** [comment]
Current status:

```sh
➜  hoverpad git:(add-eslint) $ npm test

> Hoverpad@1.7.0 test /Users/pdehaan/dev/github/mozilla/hoverpad
> npm run lint


> Hoverpad@1.7.0 lint /Users/pdehaan/dev/github/mozilla/hoverpad
> eslint .


/Users/pdehaan/dev/github/mozilla/hoverpad/encryption.js
   62:10  error  'encrypt' is defined but never used                 no-unused-vars
   66:9   error  'passphraseKey' is assigned a value but never used  no-unused-vars
  108:10  error  'decrypt' is defined but never used                 no-unused-vars
  109:9   error  'passphraseKey' is assigned a value but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla/hoverpad/ports.js
   3:7   error  'KEY_PREFIX' is assigned a value but never used  no-unused-vars
  43:44  error  'content' is defined but never used              no-unused-vars
  49:44  error  'content' is defined but never used              no-unused-vars

✖ 7 problems (7 errors, 0 warnings)


npm ERR! Darwin 16.4.0
npm ERR! argv "/Users/pdehaan/.nvm/versions/node/v6.10.0/bin/node" "/Users/pdehaan/.nvm/versions/node/v6.10.0/bin/npm" "run" "lint"
npm ERR! node v6.10.0
npm ERR! npm  v3.10.10
```

ESLint fails w/ 7 errors, but it doesn't look like we have Circle-CI hooked up to this repo and testing on each commit. Separate bug for that incoming shortly.

---

**UPDATE:** Filed _"Add Circle-CI for running tests/linting on each commit #26"_

**Comment by pdehaan** [comment]
May need to tweak the .eslintignore to include _at least_ the following:

```
electron-apps/
elm-stuff/
www/
electron.js
```

---

**UPDATE:** Added in https://github.com/mozilla-services/hoverpad/pull/25/commits/e55dd99242cce683b2ef74060d7200cd91828f31

File: encryption.js [hunk 4]
```diff
@@ -29,26 +29,26 @@ function generateKey(passphrase, appWiseSalt) {
...
-    var binary_string =  window.atob(base64);
-    var len = binary_string.length;
-    var bytes = new Uint8Array( len );
-    for (var i = 0; i < len; i++)        {
-        bytes[i] = binary_string.charCodeAt(i);
-    }
-    return bytes;
+  const binary_string =  window.atob(base64);
+  const len = binary_string.length;
+  const bytes = new Uint8Array( len );
```
Comment (magopian) [inline encryption.js]: Seems eslint doesn't check the spaces insides parens: here's there's `( len )` but in most of the other places we don't have spaces.

File: encryption.js (hunk 2 above)
Comment (magopian) [reply encryption.js]: Yeah, it's still in an alpha stage, we use many debug logs just to make it more convenient. Maybe add the exception at the top of the file, for the whole file?

File: ports.js [hunk 5]
```diff
@@ -1,6 +1,8 @@
-const KEY_PREFIX = "hoverpad";
-var app;
-if (typeof(Elm) === "undefined") {
+/* globals encrypt:false, decrypt:false */
+
+const KEY_PREFIX = 'hoverpad';
```
Comment (magopian) [inline ports.js]: Let's get rid of the dead code

**Comment by magopian** [comment]
This is very nice, thanks @pdehaan !

I'm afraid to ask the question (because of the possible bikeshedding that will follow), but should we not use `;` (and more generally, conform to [standardjs](http://standardjs.com/))? I know of at least one [mozilla project](https://github.com/browserhtml/browserhtml) that uses standardjs through [standard-flow](https://github.com/Gozala/standard-flow).

**Comment by pdehaan** [comment]
I'm indifferent on `;` vs no `;`. Personally I prefer them, but we can use whatever style guide you and your team prefers. There isn't a set of Mozilla ESLint rules that I've ever found. It usually differs between each repo/project, and in the case of FxA, even differs in the case of server vs client code.

If you prefer **standard** format, I can probably tweak the ESLint rules to extend that plugin/config's recommended rules and then reformat the code using `eslint . --fix`.

Or in some cases we just use [**prettier**](https://www.npmjs.com/package/prettier) and ESLint to handle the formatting of code.

I'm fine with any format, just as long as we're consistent, and have at least some basic ESLint config to check for unused variables, uncaught promise errors, and other basic rules.

**Comment by magopian** [comment]
Two minor comments. It all looks very good, thanks for your work @pdehaan !

File: encryption.js [hunk 6]
```diff
@@ -116,13 +121,13 @@ function decrypt(passphrase, encryptedContent) {
     .then(decryptionKey => {
       return crypto.subtle.decrypt(
         {name: 'AES-GCM',
-         iv: parts.iv},
+          iv: parts.iv},
```
Comment (Natim) [inline encryption.js]: I guess we should put name on the other line too then.

File: encryption.js (hunk 4 above)
Comment (pdehaan) [reply encryption.js]: Fixed! ESLint has a `space-in-parens` rule which let us auto-fix this in the 6 places we slipped in an extra space before/after a paren. :+1:

File: encryption.js (hunk 6 above)
Comment (pdehaan) [reply encryption.js]: Sure, I can manually change those. Not sure if ESLint has a rule for that formatting level, but another cool module we can use to format the codebase is [**prettier**](https://github.com/prettier/prettier), if we want everything converted to AST and reformatted. It's probably a fairly large diff, so I won't add that to this PR, but once this lands I can do a separate PR to see if it is worth the effort.

---

**UPDATE:** Submitted **prettier** PR as #34.

**Comment by magopian** [comment]
All good! Thank you very much!

**Pull request merged**